source/MRMesh/MRVoxelsSave.cpp

```diff
@@ -103,6 +104,107 @@ tl::expected<void, std::string> saveRAW( const std::filesystem::path& path, cons
     return {};
 }
 
+tl::expected<void, std::string> saveSliceToImage( const std::filesystem::path& path, const ObjectVoxels& voxelsObject,
+                                                             SlicePlain slicePlain, int sliceNumber, float min, float max, ProgressCallback callback/* = {} */)
+{
+    const auto& bounds = voxelsObject.getActiveBounds();
+    const auto dims = bounds.size();
+    const auto dimX = dims.x;
+    const auto dimXY = dimX * dims.y;
+    const int textureWidth = dims[( slicePlain + 1 ) % 3];
+    const int textureHeight = dims[( slicePlain + 2 ) % 3];
+
+    std::vector<Color> texture( textureWidth * textureHeight );
+    Vector3i activeVoxel;
+    switch ( slicePlain )
+    {
+    case SlicePlain::XY:
+        if ( sliceNumber > bounds.max.z )
+            return  tl::make_unexpected( "Slice number exceeds voxel object borders" );
+
+        activeVoxel = { bounds.min.x, bounds.min.y, sliceNumber };
+        break;
+    case SlicePlain::YZ:
+        if ( sliceNumber > bounds.max.x )
+            return  tl::make_unexpected( "Slice number exceeds voxel object borders" );
+
+        activeVoxel = { sliceNumber, bounds.min.y, bounds.min.z };
+        break;
+    case SlicePlain::ZX:
+        if ( sliceNumber > bounds.max.y )
+            return  tl::make_unexpected( "Slice number exceeds voxel object borders" );
+
+        activeVoxel = { bounds.min.x, sliceNumber, bounds.min.z };
+        break;
+    default:
+        return  tl::make_unexpected( "Slice plain is invalid" );
+    }
+ 
+    const auto& grid = voxelsObject.grid();
+    const auto accessor = grid->getConstAccessor();
+  
+    for ( int i = 0; i < int( texture.size() ); ++i )
+    {
+        openvdb::Coord coord;
+        coord[slicePlain] = sliceNumber;
+        coord[( slicePlain + 1 ) % 3] = ( i % textureWidth ) + bounds.min[( slicePlain + 1 ) % 3];
+        coord[( slicePlain + 2 ) % 3] = ( i / textureWidth ) + bounds.min[( slicePlain + 2 ) % 3];
+
+        const auto val = accessor.getValue( coord );
+        const float normedValue = ( val - min ) / ( max - min );
+        texture[i] = Color( Vector3f::diagonal( normedValue ) );
+
+        if ( callback )
+            callback( 1.0f / texture.size() );
```

Comment:
mb not call callback each time, better call it each 100th time, also it is always the same value now it should be 
(i/texture.size()), and callback return value is not used